tests/pythie_serving/test_wrappers.py

```diff
@@ -0,0 +1,129 @@
+import json
+import logging
+import pickle
+from unittest import TestCase
+
+import numpy as np
+import treelite_runtime
+import xgboost as xgb
+from google.protobuf import text_format
+from grpc import insecure_channel
+from numpy.testing import assert_array_almost_equal
+
+from pythie_serving.server import create_grpc_server
+from pythie_serving.tensorflow_proto.tensorflow_serving.apis import (
+    predict_pb2,
+    prediction_service_pb2_grpc,
+)
+from pythie_serving.tensorflow_proto.tensorflow_serving.config import (
+    model_server_config_pb2,
+)
+from pythie_serving.utils import make_ndarray_from_tensor, make_tensor_proto
+
+from .generate_test_resources import RESOURCES_PATH
+
+
+class WrapperTestCaseMixin:
+    model_platform = None
+    server = None
+    model = None
+    metadata = None
+
+    def setUp(self) -> None:
+
+        # serve model locally
+        model_server_config = model_server_config_pb2.ModelServerConfig()
+        text_format.Parse(
+            f"""
+        model_config_list {{
+            config {{
+                name: "{self.model_platform}-model",
+                base_path: "{RESOURCES_PATH}/{self.model_platform}-model",
+                model_platform: "{self.model_platform}"
+            }}
+        }}
+        """,
+            model_server_config,
+        )
+        self.server = create_grpc_server(
+            model_server_config=model_server_config,
+            worker_count=1,
+            port=9999,
+            maximum_concurrent_rpcs=1,
+            _logger=logging.getLogger(),
+        )
+        self.server.start()
+
+        # also deserialize the model to call directly
+        with open(f"{RESOURCES_PATH}/{self.model_platform}-model/metadata.json", "r") as f:
```

Comment:
`mode="r"` is already the default mode, so it's not necessary to specify it
Pyupgrade detected it (it is set in the pre-commit-config.yaml only, but might be useful to add it to the CI)